x-pack/plugins/apm/public/hooks/use_fetch_triggered_alert_history.ts

```diff
@@ -0,0 +1,208 @@
+        avgTimeToRecoverMS,
+      } = await fetchTriggeredAlertsHistory({
+        http,
+        index,
+        ruleId,
+        signal: abortCtrlRef.current.signal,
+      });
+
+      if (error) throw error;
+      if (!isCancelledRef.current) {
+        setTriggeredAlertsHistory((oldState: TriggeredAlertsHistory) => ({
+          ...oldState,
+          triggeredAlertsData: {
+            totalTriggeredAlerts,
+            histogramTriggeredAlerts,
+            avgTimeToRecoverMS,
+          },
+          isLoadingRuleAlertsAggs: false,
+        }));
+      }
+    } catch (error) {
+      if (!isCancelledRef.current) {
+        if (error.name !== 'AbortError') {
+          setTriggeredAlertsHistory((oldState: TriggeredAlertsHistory) => ({
+            ...oldState,
+            isLoadingRuleAlertsAggs: false,
+            errorTriggeredAlertHistory: error,
+            triggeredAlertsData: undefined,
+          }));
+        }
+      }
+    }
+  }, [features, http, ruleId]);
+  useEffect(() => {
+    loadRuleAlertsAgg();
+  }, [loadRuleAlertsAgg]);
+
+  return triggeredAlertsHistory;
+}
+
+interface IndexName {
+  index: string;
+}
+
+export async function fetchIndexNameAPI({
+  http,
+  features,
+}: {
+  http: HttpSetup;
+  features: string;
+}): Promise<IndexName> {
+  const res = await http.get<{ index_name: string[] }>(
+    `${BASE_RAC_ALERTS_API_PATH}/index`,
+    {
+      query: { features },
+    }
+  );
+  return {
+    index: res.index_name[0],
+  };
+}
+
+export async function fetchTriggeredAlertsHistory({
+  http,
+  index,
+  ruleId,
+  signal,
+}: {
+  http: HttpSetup;
+  index: string;
+  ruleId: string;
+  signal: AbortSignal;
+}): Promise<FetchTriggeredAlertsHistory> {
+  try {
+    const res = await http.post<AsApiContract<any>>(
+      `${BASE_RAC_ALERTS_API_PATH}/find`,
+      {
+        signal,
+        body: JSON.stringify({
+          index,
+          size: 0,
+          query: {
+            bool: {
+              must: [
+                {
+                  term: {
+                    [ALERT_RULE_UUID]: ruleId,
+                  },
+                },
+                {
+                  range: {
+                    [ALERT_TIME_RANGE]: {
+                      gte: 'now-30d',
+                      lt: 'now',
+                    },
+                  },
+                },
+              ],
+            },
+          },
+          aggs: {
+            histogramTriggeredAlerts: {
+              date_histogram: {
+                field: ALERT_START,
+                fixed_interval: '1d',
+                extended_bounds: {
+                  min: 'now-30d',
+                  max: 'now',
+                },
+              },
+            },
+            avgTimeToRecoverMS: {
+              avg: {
+                script: {
+                  source:
+                    "if (!doc['kibana.alert.end'].empty){return(doc['kibana.alert.end'].value.millis) - (doc['kibana.alert.start'].value.millis)}",
```

Comment:
I thought you updated this?

Reply:
Yeah, I did, and I updated the type for the `find` API to accept that update. I don't know why this script is still there 😕. Maybe it was lost when I moved the hook from Observability to APM

Reply:
fixed